2015/presentations/with modern_peripherals: Python and Flask Piotr Dyba/text.md

```diff
@@ -0,0 +1,210 @@
+    emit('msg_event', msg)
+```
+
+The front end is an out-of-the-box solution and usage is as simple as attaching 
+single \<script\> tag library. When using with Flask-SocketIO we have to use 
+0.9.16 version of JS SocketIO library, because of incompatibility with 1.0.
+
+Frontend code:
+```javascript
+$(document).ready(function() {
+    var namespace = '/test';
+    var socket = io.connect('http://' + document.domain + ':' + location.port + namespace);
+    // connecting event
+    socket.on('connect', function () {
+        socket.emit('connect');
+        $('#status').append('connected');
+    });
+    // event on frontend activated from backend
+    socket.on('even_name', function (){
+        console.log("hi");
+        return 0;
+    });
+    // event in backend activated from frontend
+    $('#some_button').click(function() {
+        socket.emit('from_end_event');
+        return 0;
+    });
+});
+```
+
+##Eye Tribe
+
+EyeTribe is an eye tracker hardware that is advertised as a cheap solution, 
+despite the self-describing name, what is eye tracking?
+Eye tracking is the process of using sensors to estimate where someone 
+is looking – a point of gaze. Eye tribe uses infrared sensors for that purpose.
+Eye tracking can be used in a wide variety of applications typically 
+categorized as active or passive. Active applications involve device control, 
+for example aiming in games, eye activated login or hands-free typing. 
+Even in ’80 and ‘90 there were a few photo cameras that used it, like Canon 1V for
+choosing the camera's focus point. 
+Passive applications include performance analysis of design, 
+layout and advertising. The main usage of the eye tracking for web developers 
+is analysis of the web app that will improve ads, layout and design.
+Controlling the site with eyes will be the next step. It will be especially useful 
+for handicapped people with even severe body or mind damage. Think about 
+Steven Hawking, and how well he is doing despite his condition, but using eye 
+tracking in such a manner needs some level of focus. I think that when eye 
+trackers will get more popular, or using the built-in cameras for this purpose 
+will be more efficient and common… when we will get used to the technology,
+its advantages and some of people will start using it on daily basis, the eye 
+following ads will emerge. Closed only by blinking three times or roll eyes gestures
+
+#### Wrapper
+
+The Python wrapper is straightforward and one file only. 
+[Eye Tribe wrapper](https://github.com/baekgaard/peyetribe)
+It connects directly to the Eye Tribe daemon. The Eye Tribe was the easiest 
+device to use with Python. The wrapper provides only information about point of gaze and
+ eyes location for each eye separately, but that is all what we need.
+
+#### JS vs Python
+I've compared efficiency of Python solution against pure JavaScript one, during
+two tests, the first app types one for each approach was displaying live data 
```

Comment:
the first ~~types~~ one was displaying live data for each approach, ~~and~~ the soecond was only collecting data.